ini/teensy.ini

```diff
@@ -9,61 +9,71 @@
 #                               #
 #################################
 
+#
+# Teensy AVR
+#
+[teensy_avr]
+extends    = common_avr8
+platform   = teensy
+lib_ignore = ${env:common_avr8.lib_ignore}, NativeEthernet
+
 #
 # AT90USB1286 boards using CDC bootloader
 # e.g., BRAINWAVE, BRAINWAVE_PRO, SAV_MKI, TEENSYLU
 #
 [env:at90usb1286_cdc]
-platform   = teensy
-extends    = common_avr8
-board      = marlin_at90usb1286
-lib_ignore = ${env:common_avr8.lib_ignore}, Teensy_ADC, NativeEthernet
+extends           = teensy_avr
+platform_packages = toolchain-atmelavr@~1.70300.0
```

Comment:
You should move the platform_packages override up into your new teensy_AVR section.